Testing/Readme.md

```diff
@@ -0,0 +1,315 @@
+
+The checklist below should be used by the development team when it adds a new feature or updates an existing policy.  The goal of the list below is to ensure consistency of automated functional testing and overall quality of ScubaGear.
+
+- [ ] Ensure all Non-Complaint configuration options for the affected policy are tested.
+- [ ] Ensure all Complaint configuration options for the affected policy are tested.
+- [ ] Ensure that any changes to HTML report output are tested.
+- [ ] Validate that all new functional tests pass on CI for the branch before creating the PR
+
+
+### Adding new functional test - Example #1 ###
+
+Use case: functional test for a SCB where there is a single non-complaint configuration option and a single complaint configuration option.
+
+Example SCB: MS.POWERPLATFORM.1.1v1 - the ability to create production and sandbox environments should be restricted to admin users
+
+Configuration options: looking into the tenant, there are only two configurations options for restricting the ability to create production and sandbox environments. Non-complaint config: allowing everyone to create production and sandbox environments. Complaint config: restricting it only specific admins.
+
+![Power Platform 1.1 Config](/images/PP-1-1.png)
+
+Automated functional test case for non-complaint and complaint configs:
+```
+  - PolicyId: MS.POWERPLATFORM.1.1v1
+    TestDriver: RunScuba
+    Tests:
+      - TestDescription: MS.POWERPLATFORM.1.1v1 Non-Compliant case - Allows production/sandbox environment creation by nonadmin users
+        Preconditions:
+          - Command: Set-TenantSettings
+            Splat:
+              RequestBody:
+                DisableEnvironmentCreationByNonAdminUsers: false
+        Postconditions: []
+        ExpectedResult: false
+      - TestDescription: MS.POWERPLATFORM.1.1v1 Compliant case - Restricts production/sandbox environment creation to admin users
+        Preconditions:
+          - Command: Set-TenantSettings
+            Splat:
+              RequestBody:
+                DisableEnvironmentCreationByNonAdminUsers: true
+        Postconditions: []
+        ExpectedResult: true
+```
+
+### Adding new functional test - Example #2 ### 
+
+Use case: functional test for a SCB where there are multiple non-complaint configuration options.
+
+Example SCB: MS.TEAMS.1.3v1: restricting dial-in users and anonymous users from being admitted into a Teams meeting automatically.
+
+Configuration options: looking into the tenant there are multiple options in allowing and restricting different groups of users to automatically admitted into a meeting. There are two config options that would fail the SCB and others that would pass the SCB.
```

Comment:
```suggestion
**Configuration options**: Looking into the tenant, there are multiple options for allowing and restricting different groups of users to be automatically admitted into a meeting. There are two config options that would fail the SCB, and others that would pass the SCB.
```